### PR TITLE
Document and Refactor Compiler Exceptions

### PR DESCRIPTION
**Pull request by dsouzai**
Addresses https://github.com/eclipse/openj9/issues/163

### CODE REVIEW

**Comment by dsouzai** [comment]
@mpirvu could you please review.

File: runtime/compiler/trj9/control/CompilationThread.cpp [hunk 1]
```diff
@@ -9737,6 +9736,9 @@ TR::CompilationInfoPerThreadBase::processException(
       throw;
       }
 
+   // The catch blocks below are ordered such that in each group of
+   // blocks, the basest type is at the end of the group.
```
Comment (mgaudet) [inline runtime/compiler/trj9/control/CompilationThread.cpp]: Maybe replace "basest" with least-derived, or most generic?

File: runtime/compiler/trj9/exceptions/AOTFailure.hpp [hunk 2]
```diff
@@ -26,25 +26,46 @@
...
+ */
 class AOTHasInvokeVarHandle : public virtual TR::RecoverableILGenException
    {
    virtual const char* what() const throw() { return "AOT Has Invoke VarHandle"; }
    };
 
+/**
+ * AOT Has Invoke Special in Interface exception type.
+ *
+ * Thrown when a method that has an invoke space in an interface is AOT Compiled.
```
Comment (mpirvu) [inline runtime/compiler/trj9/exceptions/AOTFailure.hpp]: Replace "invoke space" with "invoke special"

**Comment by dsouzai** [comment]
@mpirvu I made the request change. I've also run pbuild tests. This should be ok to merge as is, or if you'd like, after a Jenkins Sanity test.

**Comment by mpirvu** [comment]
jenkins test sanity

**Pull request merged**